### PR TITLE
Tech Debt cleanup: move LogOnError into common datastore package

### LINKED ISSUE #609: Move LogOnError from migrations package in MySQL into the main package
Its a nice utility and not only used by migrations anymore

### PR DESCRIPTION
**Pull request by josephschorr**
Fixes #609

### CODE REVIEW

File: internal/datastore/common/logging.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"context"
+
+	log "github.com/authzed/spicedb/internal/logging"
+)
+
+// LogOnError executes the function and logs the error.
+// Useful to avoid silently ignoring errors in defer statements
+func LogOnError(ctx context.Context, f func() error) {
+	if err := f(); err != nil {
+		log.Ctx(ctx).Error().Err(err).Msg("datastore error")
```
Comment (vroldanbet) [inline internal/datastore/common/logging.go:13]: `Error()` here is redundant. From zerolog API:

```go
// Err starts a new message with error level with err as a field if not nil or
// with info level if err is nil.
//
// You must call Msg on the returned event in order to send the event.
func (l *Logger) Err(err error) *Event {
	if err != nil {
		return l.Error().Err(err)
	}

	return l.Info()
}
```

**Review by vroldanbet — APPROVED** [review]
LGTM, just a small nit

File: internal/datastore/common/logging.go (hunk 1 above)
Comment (josephschorr) [reply internal/datastore/common/logging.go:13]: That's what I get for copy-pasting :)

Comment (josephschorr) [reply internal/datastore/common/logging.go]: Fixed

**Review by vroldanbet — APPROVED** [review]

**Pull request merged**